pkg/watch/interface.go

```diff
@@ -3,10 +3,14 @@ package watch
 import (
 	"context"
 	"io"
+
+	"github.com/devfile/library/pkg/devfile/parser"
 )
 
 type Client interface {
 	// WatchAndPush watches the component under the context directory and triggers Push if there are any changes
 	// It also listens on ctx's Done channel to trigger cleanup when indicated to do so
 	WatchAndPush(out io.Writer, parameters WatchParameters, ctx context.Context) error
+	// CleanupFunc deletes the component created using the devfileObj and writes any outputs to out
+	CleanupFunc(devfileObj parser.DevfileObj, out io.Writer) error
```

Comment:
You'll need to update the mock client after this change.

Reply:
7b82c4a Apparently there were other mocks as well that got updated. 😂

Reply:
Isn't the `Func` suffix redundant? This is already a method. :)
I think the `Func` suffix would have made sense if this was returning a function, but this is not the case here.

Reply:
Makes sense. Removed.